src/services/swap/swap.service.ts

```diff
@@ -0,0 +1,349 @@
+
+export enum SwapTokenType {
+  fixed,
+  min,
+  max
+}
+
+export interface SwapToken {
+  address: Address;
+  amount: BigNumber;
+  type: SwapTokenType;
+}
+
+export default class SwapService {
+  constructor(
+    private readonly config: ConfigService = configService,
+    private readonly web3: Web3Service = web3Service
+  ) {}
+
+  public async batchSwapV1(
+    tokenIn: SwapToken,
+    tokenOut: SwapToken,
+    swaps: Swap[][]
+  ): Promise<TransactionResponse> {
+    console.log('[Swap Service] batchSwapV1');
+
+    const options: any = {};
+
+    if (tokenIn.address === NATIVE_ASSET_ADDRESS) {
+      options.value = tokenIn.amount;
+    }
+
+    try {
+      return exchangeProxyService.multihopBatchSwap(
+        swaps,
+        tokenIn,
+        tokenOut,
+        options
+      );
+    } catch (e) {
+      console.log('[Swapper] batchSwapV1 Error:', e);
+      return Promise.reject(e);
+    }
+  }
+
+  public async batchSwapV2(
+    tokenIn: SwapToken,
+    tokenOut: SwapToken,
+    swaps: SwapV2[],
+    tokenAddresses: string[]
+  ): Promise<TransactionResponse> {
+    if (isStETH(tokenIn.address, tokenOut.address)) {
+      return this.lidoBatchSwap(tokenIn, tokenOut, swaps, tokenAddresses);
+    }
+    console.log('[Swap Service] batchSwapV2');
+    const overrides: any = {};
+
+    if (tokenIn.address === AddressZero) {
+      overrides.value = tokenIn.amount;
+    }
+
+    const swapKind =
+      tokenOut.type === SwapTokenType.min
+        ? SwapKind.GivenIn
+        : SwapKind.GivenOut;
+
+    const funds = await this.getFundManagement();
+
+    try {
+      if (swaps.length == 1) {
+        const single: SingleSwap = {
+          poolId: swaps[0].poolId,
+          kind: swapKind,
+          assetIn: tokenAddresses[swaps[0].assetInIndex],
+          assetOut: tokenAddresses[swaps[0].assetOutIndex],
+          amount: swaps[0].amount,
+          userData: swaps[0].userData
+        };
+
+        return vaultService.swap(
+          single,
+          funds,
+          tokenOut.amount.toString(),
+          overrides
+        );
+      }
+
+      const limits: string[] = this.calculateLimits(
+        tokenIn,
+        tokenOut,
+        tokenAddresses
+      );
+
+      return vaultService.batchSwap(
+        swapKind,
+        swaps,
+        tokenAddresses,
+        funds,
+        limits,
+        overrides
+      );
+    } catch (e) {
+      console.log('[Swapper] batchSwapV2 Error:', e);
+      return Promise.reject(e);
+    }
+  }
+
+  public async lidoBatchSwap(
+    tokenIn: SwapToken,
+    tokenOut: SwapToken,
+    swaps: SwapV2[],
+    tokenAddresses: string[]
+  ): Promise<TransactionResponse> {
+    console.log('[Swapper] lidoBatchSwap');
+    const overrides: any = {};
+
+    if (tokenIn.address === AddressZero) {
+      overrides.value = tokenIn.amount;
+    }
+
+    // Convert tokenIn/tokenOut so that it matches what's in tokenAddresses
+    const { stETH, wstETH } = this.config.network.addresses;
```

Comment:
minor - I think it would nice if was `stETHaddr` or even using `this.config.network.addresses.stETH` because `stETH.toLowerCase()` reads as if this was a symbol